[CS2113-T13-3] WhereGotTime

yeapcl opened the pull request:
An app that detects time table clashes

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
+while the other two will result in the app giving an error message
+prompting the user to re-enter again.<br/>
+
+### Edit Function<br/>
```
This segment seems incomplete? Minimally some description of what these functions are instead of simply listing would be helpful to the reader.

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
+#### Design of the Add Function<br/>
+
+UML Class Diagram of the Add function:
+![](team/uml.PNG)<br/>
```
The UML diagram looked a bit strange with the large blank spaces within the classes and the much smaller font used for the multiplicities. 

![image](https://user-images.githubusercontent.com/53657436/97386254-659da980-190e-11eb-8e84-b07ae86fb4d5.png)

AmeliaTYR replied:
The filled in black arrow heads were to be avoided iirc? (inheritance or association issue?)

![image](https://user-images.githubusercontent.com/53657436/97386314-89f98600-190e-11eb-9fde-4327bf33a22c.png)

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
+and location) to the timetable class into its arraylist divided by the day of the week<br/>
+
+#### Sequence Design of Add Function<br/>
+![](team/seqdiagram.jpg)<br/>
```
![image](https://user-images.githubusercontent.com/53657436/97386637-350a3f80-190f-11eb-81aa-6903024aa521.png)

You could perhaps indicate more clearly where the arrows are coming from or going.
For the sake of neatness you could try to do the diagrams in software like powerpoint some or uml creator or something instead of writing cause the pencil marks here are a bit messy sorry

amalinasani replied:
You could also perhaps use the same software to create your uml diagrams for the sake of consistency, as the diagrams vary in style along the document.

amalinasani replied:
![image](https://user-images.githubusercontent.com/60027323/97515331-cf2fbd80-19cb-11eb-9874-45422dfea44f.png)
1. parsedInputs[] cannot be a method call. 
2. "Return" lines should orginate from the point of deactivation
3. "new event" method call line should not point backwards in time. Perhaps you can reformat this diagram such that it is a straight line.

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
 
+#### System Architecture of the Add Function<br/>
+
+![](team/Architect-digram.PNG)
```
For this diagram it looks like the user can by pass the interface and directly add commands? Its a bit confusing whether this was intended or why it was designed this way. Further explanation on this part of the diagram would be helpful

![image](https://user-images.githubusercontent.com/53657436/97387037-2708ee80-1910-11eb-9a5a-c944017f6abf.png)

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](team/CompareCommand_ClassDiagram.jpg) <br/>
```
![image](https://user-images.githubusercontent.com/53657436/97387108-4bfd6180-1910-11eb-95d5-de34d35fd8ee.png)

this diagram faces a similar issue of why the user can by pass the ui as mentioned above

amalinasani replied:
Same issue with the second "blue" user.

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](team/CompareCommand_ClassDiagram.jpg) <br/>
```
![image](https://user-images.githubusercontent.com/53657436/97387166-72230180-1910-11eb-8d8c-f224cd21e827.png)

For this diagram why are there two different coloured users and why do they have different access to the program? The purpose of drawing this additional user and their role is unclear, even in the explanation below.

The duplicated timetable box is also confusing.

AmeliaTYR commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,138 @@
+
+
+#### Sequence Design of Compare Function<br/>
+![](team/CompareCommand_SequenceDesign.jpg) <br/>
```
![image](https://user-images.githubusercontent.com/53657436/97388356-dfd02d00-1912-11eb-9aba-ccc3d80b50e7.png)

For this diagram it is a bit disorganized and confusing.

Firstly, it is a bit unclear whether the User class here refers to the User class in the compare function (in which case there shouldn't be a random method call coming from it) or if it refers to an actual user (in which case the notation should be the stick figure not the class header)

Secondly, from the CompareCommand activation bar has an unknown return coming from the middle of the activation bar, while the parsedInputs method call being made to the same activation bar is too low down. This is also the case with the activation bar for the self invoked method call execute and the common time array method call to the UI class. 

It is also unclear where the method call to the UI intends to return to as it is chronologically further down than the end of the return from the parsedInputs method.

AmeliaTYR reviewed (commented):
I like that a lot of effort was put into making several diagrams explaining various components of the program and the aesthetic of some of the digitally hand drawn diagrams, though further elaboration will help enhance the usefulness of your diagrams and help the user better understand your code and implementation rationale. Good work!

arindamshivatrip commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,150 @@
+|v2.0|user|be assured that my inputs are correctly added|my input are correctly convey into the timetable|
+|v2.0|user|access my saved timetables|I do not have to manually enter the timetables again|
 
 ## Non-Functional Requirements
```
This part was  left empty. 
![image](https://user-images.githubusercontent.com/47693437/97457993-3b330700-1975-11eb-917b-835aa5b1aca1.png)

arindamshivatrip commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,150 @@
+
+
+#### Sequence Design of Compare Function<br/>
+![](team/CompareCommand_SequenceDesign.jpg) <br/>
```
![image](https://user-images.githubusercontent.com/47693437/97458101-569e1200-1975-11eb-942a-8944348e0e05.png)
Here there is a significant lack of explaination given and the diagram appears to be slightly messy and confounding.  For example what is the user written here. Is it a class or the user operating the program.

arindamshivatrip commented on docs/DeveloperGuide.md:
```diff
@@ -2,24 +2,150 @@
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](team/CompareCommand_ClassDiagram.jpg) <br/>
```
![image](https://user-images.githubusercontent.com/47693437/97458497-bb596c80-1975-11eb-87b3-99338275416a.png)
Here is there supposed to be a chronological order to the diagram that I haven't noticed.?
 If not then the 2 users and 2 timetables confound the reader.

arindamshivatrip reviewed (commented):
ON the whole the diagrams are a little confusing. but well made and visually appealing.

amalinasani commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,168 @@
+* To ensure that the users do not enter duplicate event that has the same timing or in between the time 
+that is entered. 
+* A verification check is added to the add command of the version 2.0 of WhereGotTime.
+
```
You could perhaps add a segment to explain the above sequence diagram?

amalinasani commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,168 @@
+
+#### Sequence Diagram
+![](team/LogInCommand_Sequence_v002.png)
+
```
Missing colons for all objects
![image](https://user-images.githubusercontent.com/60027323/97514844-b07cf700-19ca-11eb-8a6f-e80988714f82.png)

yuxinwu99 replied:
Opt should be used instead of alt here as there are no alternatives provided.

tammykoh committed:
Updated UG - included design for delete, clear,  list and find functions

tammykoh committed:
Updated DG - included sequence diagrams for delete, clear, list and find functions

tammykoh committed:
Merge branch 'tammy-developer-guide'

yeapcl committed:
updated DG

tammykoh committed:
Updated UG - renamed heading

yeapcl committed:
updated DG

yeapcl committed:
Merge branch 'master' into brach-yeapcl-Documentation

* master:
  updated DG
  updated DG
  Bug Fix to Add command

yeapcl committed:
Merge pull request #110 from yeapcl/brach-yeapcl-Documentation

Brach yeapcl documentation

Lusi711 commented on docs/DeveloperGuide.md:
```diff
@@ -31,4 +174,27 @@
 
 ## Instructions for manual testing
```
Is it better to follow the format given in [here](https://se-education.org/addressbook-level3/DeveloperGuide.html#appendix-instructions-for-manual-testing)?

tammykoh committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/images/Overall Architecture.png
#	docs/team/Overall Architecture.png
#	docs/team/OverallArchi.png

tammykoh committed:
Merge branch 'master' of https://github.com/tammykoh/tp

tammykoh committed:
Deleted unused line in ClearCommand

yeapcl committed:
Merge pull request #112 from tammykoh/master

Updated DG

wangwaynesg commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+and location) to the timetable class into its arraylist divided by the day of the week<br/>
+
+#### Sequence Design of Add Function<br/>
+![](images/seqdiagram.jpg)<br/>
```
Perhaps this sequence diagram should be done using an online application instead of hand drawn?

TanJunHong replied:
Would it be better to use a software to draw the sequence diagram instead of hand drawing it?

yuxinwu99 replied:
There should be a return arrow from the activation bar under the Timetable class to the activation bar of execute().

wangwaynesg commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+* UserList: The Class that creates an ArrayList where the User objects will be stored
+
+#### Class Diagram
+![](images/LogInCommand_Class_Diagram_v002.png)
```
Perhaps some methods like getters should be omitted from the class diagram to simplify and increase readability of the class diagram? 😄

wangwaynesg reviewed (commented):
Very nice developers guide! It was hard to find things to comment about!

TanJunHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+
+The following Sequence Diagram displays how components interact when the user inputs `clear /fri`
+
+![Architecture Sequence Diagram](images/ArchitectureSequenceDiagram.png)
```
This image is not loading for me. Perhaps the ".png" is supposed to be ".PNG" instead?

TanJunHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+
+#### Design of the Delete Function<br/>
+
+![](images/DeleteCommand_umlDiagram.png)
```
Is there a better way to show 3 arrows pointing to Ui? Perhaps a slightly larger diagram would help?

TanJunHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+
+### Clear Function
+
+#### System Architecture of the Clear Function<br/>
```
Perhaps the system architecture of the delete and clear functions could be consolidated, since they are similar in nature?

TanJunHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+* UserList: An arraylist of User<br/>
+
+#### Design of the Compare Function<br/>
+![](images/CompareCommand_UML_Diagram.jpg)<br/>
```
Is there a missing bracket in the getTimetable function?

TanJunHong commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+#### Design of the Add Function<br/>
+
+UML Class Diagram of the Add function:
+![](images/uml.PNG)<br/>
```
Is the bracket of the execute function supposed to be beside "execute" instead of "void"?

TanJunHong reviewed (commented):
Overall, it looks good!
There might be some mistakes here and there.

wly99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+
+#### System Architecture of the List Function<br/>
+
+![](images/ListCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the list command.<br/>
+* UI: The User Interface of the app<br/>
+* ListCommand: The main logic command of the list function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the List Function<br/>
+
+![](images/ListCommand_umlDiagram.png)
```
<img width="757" alt="Screenshot 2020-10-29 at 4 36 24 PM" src="https://user-images.githubusercontent.com/34613421/97544671-e63ed180-1a04-11eb-9b0d-dc706c1604b2.png">
The images show arrows pointing in different directions between ListCommand and Timetable, does this matter?

wly99 replied:
<img width="750" alt="Screenshot 2020-10-29 at 4 39 28 PM" src="https://user-images.githubusercontent.com/34613421/97544998-53526700-1a05-11eb-903a-a1527e46fc47.png">
class diagram is a little confusing in the relationships, does LIstCommand "users" or "uses" UI?

wly99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+1. `ListCommand` then passes the user's timetable(s) to the `UI`
+1. `UI` prints all the classes in the timetable
+
+#### Sequence Design of List Function<br/>
+![](images/ListCommand_SequenceDiagram.png)<br/>
```
<img width="605" alt="Screenshot 2020-10-29 at 4 41 08 PM" src="https://user-images.githubusercontent.com/34613421/97545174-8e549a80-1a05-11eb-8bf9-d88eda3a1655.png">
Does it matter if the first arrow pointing to ListCommand is a variable or method call?

yuxinwu99 replied:
Might be better to add  an activation bar under the Ui class when printList is called. After calling the printList method, arrows can then be used to indicate returning from Ui to ListCommand and also from execute() to the ListCommand class itself

wly99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+1. `FindCommand` passes the matching events to `UI`
+1. `UI` prints the matching events
+
+#### Sequence Design of Find Function<br/>
+![](images/FindCommand_SequenceDiagram.png)<br/>
```
<img width="596" alt="Screenshot 2020-10-29 at 4 43 37 PM" src="https://user-images.githubusercontent.com/34613421/97545395-e8556000-1a05-11eb-8b92-bf57af3301f7.png">
Does it matter if a variable was passed to FindCommand or should it be a method call with the variable inside?

yuxinwu99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,25 +1,290 @@
+timeslots the User and Target User have on a specific day of the week.<br/>
+
+
+#### Sequence Design of Compare Function<br/>
+![](images/CompareCommand_SequenceDesign.jpg) <br/>
```
You can improve this by maybe making the arrow that indicate returning from the Ui class from the end of the activation bar, it should reach the activation bar of the execute() in CompareCommand class. Also, try to use puml for diagrams as it would be neater compared to drawing it by hand.

yeapcl committed:
updated UG

yeapcl committed:
Merge pull request #119 from yeapcl/brach-yeapcl-Documentation

updated UG

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  updated UG
  Deleted unused line in ClearCommand
  Updated UG - renamed heading
  Updated DG - included sequence diagrams for delete, clear, list and find functions
  Updated UG - included design for delete, clear,  list and find functions
  Updated UG - created UML Diagrams for clear, delete, find and list commands
  Update DG - included system architecture for delete, clear, list and find functions
  Updated DG 'Architecture' section
  Changed 'Duke' to 'WhereGotTime'
  Updated 'Setting Up' section in DG
  Created folder for DG images

manuelmanuntag96 committed:
Fixed the crash in login when username/password is inputted. now throws an exeption to remind users to follow proper format login username /password.

manuelmanuntag96 committed:
Fixed the error where users can input just whitespaces. Now throws an exception telling users that the username space is empty. Cheeky buggers

yeapcl committed:
Merge pull request #164 from manuelmanuntag96/Manuel-LogInErrorFix

Manuel log in error fix

yeapcl committed:
Merge pull request #165 from manuelmanuntag96/Manuel-EmptyStringFix

Manuel empty string fix

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Fixed the error messages that show inaccurate messages. Now made clearer.

manuelmanuntag96 committed:
Fixed the bug where special characters are not sanitised and would break the login feature. Now throws an exception saying : "Username needs to contain one word with only Alphabets"

yeapcl committed:
Merge pull request #168 from manuelmanuntag96/Manuel-BugFixes

Manuel bug fixes

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Fixed the bug where special characters are not sanitised and would break the login feature. Now throws an exception saying : "Username needs to contain one word with only Alphabets"
  Fixed the error messages that show inaccurate messages. Now made clearer.
  Fixed the error where users can input just whitespaces. Now throws an exception telling users that the username space is empty. Cheeky buggers
  Fixed the crash in login when username/password is inputted. now throws an exeption to remind users to follow proper format login username /password.

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Fixed the bug where special characters are not sanitised and would break the login feature. Now throws an exception saying : "Username needs to contain one word with only Alphabets"
  Fixed the error messages that show inaccurate messages. Now made clearer.
  Fixed the error where users can input just whitespaces. Now throws an exception telling users that the username space is empty. Cheeky buggers
  Fixed the crash in login when username/password is inputted. now throws an exeption to remind users to follow proper format login username /password.

yeapcl committed:
Deep-rename to replace 'duke' with 'WhereGotTime'
Renamed seedu.duke folder to seedu.
Renamed DukeTest to WhereGotTimeTest.

yeapcl committed:
updated build.gradle for renaming duke to WhereGotTime

yellow-fellow committed:
Merge pull request #169 from yeapcl/branch-yeapcl-enhanceProgramUsability

Deep rename project from Duke to WhereGotTime

yellow-fellow committed:
Updated documentation bugs

yellow-fellow committed:
Update UserGuide.md

yellow-fellow committed:
Merge branch 'master' into master

yellow-fellow committed:
Merge pull request #170 from yellow-fellow/master

Updated documentation bugs

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Update UserGuide.md
  Updated documentation bugs
  updated build.gradle for renaming duke to WhereGotTime
  Deep-rename to replace 'duke' with 'WhereGotTime' Renamed seedu.duke folder to seedu. Renamed DukeTest to WhereGotTimeTest.

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Update UserGuide.md
  Updated documentation bugs

yeapcl committed:
Merge branch 'master' into branch-yeapcl-Documentation

* master:
  Update UserGuide.md
  Updated documentation bugs
  updated build.gradle for renaming duke to WhereGotTime
  Deep-rename to replace 'duke' with 'WhereGotTime' Renamed seedu.duke folder to seedu. Renamed DukeTest to WhereGotTimeTest.
  Fixed the bug where special characters are not sanitised and would break the login feature. Now throws an exception saying : "Username needs to contain one word with only Alphabets"
  Fixed the error messages that show inaccurate messages. Now made clearer.
  Fixed the error where users can input just whitespaces. Now throws an exception telling users that the username space is empty. Cheeky buggers
  Fixed the crash in login when username/password is inputted. now throws an exeption to remind users to follow proper format login username /password.
  Deleted unused line in ClearCommand
  Updated UG - renamed heading
  Updated DG - included sequence diagrams for delete, clear, list and find functions
  Updated UG - included design for delete, clear,  list and find functions
  Updated UG - created UML Diagrams for clear, delete, find and list commands
  Update DG - included system architecture for delete, clear, list and find functions
  Updated DG 'Architecture' section
  Changed 'Duke' to 'WhereGotTime'
  Updated 'Setting Up' section in DG
  Created folder for DG images

Commit:
Bug Fix for V2.0

yellow-fellow committed:
Merge pull request #171 from weisiong24/BugFixes

Bug Fix for V2.0

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Bug Fix for V2.0

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Bug Fix for V2.0

yeapcl committed:
changed exception naming to WhereGotTimeException

yeapcl committed:
1. Command is no longer case sensitive
2. Added Help command for easy overview of the various commands and correct input formats.
3. Help command allows user to know if they're logged in.

yeapcl committed:
Edit command no longer accepts input if user does not log in to WhereGotTime

yeapcl committed:
fixed checkstyleMain

tammykoh committed:
Merge pull request #174 from yeapcl/branch-yeapcl-enhanceProgramUsability

Branch yeapcl enhance program usability

yeapcl committed:
replaced all duke occurrences to WhereGotTime

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  fixed checkstyleMain
  Edit command no longer accepts input if user does not log in to WhereGotTime
  1. Command is no longer case sensitive 2. Added Help command for easy overview of the various commands and correct input formats. 3. Help command allows user to know if they're logged in.
  changed exception naming to WhereGotTimeException

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:

Commit:
Minor Bug fixes

yeapcl committed:
Merge pull request #175 from weisiong24/BugFixes

Minor Bug fixes

manuelmanuntag96 committed:
Edited the format for the login command
- now similar to all the other commands

Started working on the Remove User Command [Work In Progress]

manuelmanuntag96 committed:
corrected the runtest.bat expected outputs

manuelmanuntag96 committed:
Corrected the parse_loginCommand_parsedCorrectly()

yeapcl committed:
Merge pull request #178 from manuelmanuntag96/Manuel-NewLoginFormat

Manuel new login format

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Corrected the parse_loginCommand_parsedCorrectly()
  corrected the runtest.bat expected outputs
  Edited the format for the login command - now similar to all the other commands
  Minor Bug fixes

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Corrected the parse_loginCommand_parsedCorrectly()
  corrected the runtest.bat expected outputs
  Edited the format for the login command - now similar to all the other commands
  Minor Bug fixes

yeapcl committed:
1. Fixed bugs in EditCommand where the program accepts invalid timings such as 1080-1199; the timing should be in 24-hour format
2. Added more informative and user-friendly greeting messages
3. Slightly modified test cases to prove command is case-insensitive

weisiong24 committed:
Merge pull request #180 from yeapcl/branch-yeapcl-enhanceProgramUsability

Invalid time fix for EditCommand and other improvements

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  1. Fixed bugs in EditCommand where the program accepts invalid timings such as 1080-1199; the timing should be in 24-hour format 2. Added more informative and user-friendly greeting messages 3. Slightly modified test cases to prove command is case-insensitive
  replaced all duke occurrences to WhereGotTime

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:

Commit:
Bug Fixes for v2.1

yeapcl committed:
1. Bug fix for EditCommand
2. Added more "help" messages for user

yeapcl committed:
fixed checkstyle error

weisiong24 committed:
Merge pull request #182 from yeapcl/branch-yeapcl-enhanceProgramUsability

further bug fixes for EditCommand

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  fixed checkstyle error
  1. Bug fix for EditCommand 2. Added more "help" messages for user

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:

yeapcl committed:
Merge pull request #181 from weisiong24/BugFixes

Bug Fixes for v2.1

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Bug Fixes for v2.1

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Bug Fixes for v2.1

yeapcl committed:
bug fix where user is allowed to perform edit and compare commands without logging in

yeapcl committed:
Merge pull request #183 from yeapcl/branch-yeapcl-enhanceProgramUsability

Bug fix for edit and compare commands when user is not logged in

yeapcl committed:
improved EditCommand where if new timings and original timings are identical, program would prompt user.

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  bug fix where user is allowed to perform edit and compare commands without logging in

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:

weisiong24 committed:
Merge pull request #184 from yeapcl/branch-yeapcl-enhanceProgramUsability

Improved EditCommand to identify identical timings

Commit:
Updated Expected outputs for Edit,list,login,find,delete,clear,exit Commands in the Userguide

Commit:
Updates to Userguid expected outputs and command formats corrections

yeapcl committed:
Merge pull request #186 from weisiong24/BugFixes

Updated Expected outputs for Commands in Userguide

manuelmanuntag96 committed:
Implemented the Remove Feature

format: remove /username

yeapcl committed:
Merge pull request #187 from manuelmanuntag96/Manuel-RemoveFunction

Implemented the Remove Feature

Commit:
Updates to UserGuide

weisiong24 committed:
Merge pull request #188 from weisiong24/BugFixes

Updates to UserGuide

Commit:
Update to UserGuide command summary

weisiong24 committed:
Merge pull request #189 from weisiong24/BugFixes

Update to UserGuide command summary

Commit:
Updates to Developer Guide

weisiong24 committed:
Merge pull request #190 from weisiong24/BugFixes

Updates to Developer Guide

weisiong24 committed:
Update UserGuide.md

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

Commit:
Updates the About Us and the table of content for User Guide

weisiong24 committed:
Merge pull request #191 from weisiong24/BugFixes

Updates the About Us and the table of content for User Guide

manuelmanuntag96 committed:
Added more test cases for UserListTest.java

100% Branch Coverage

Commit:
Updated Userguide and add more Junit testing

Commit:
Updates to User Guide and added more Junit test

weisiong24 committed:
Merge pull request #193 from weisiong24/BugFixes

Updated UserGuide and Junit Test

manuelmanuntag96 committed:
Added more test cases for parse login and remove validity in ParserTest.java

manuelmanuntag96 committed:
Added even more test cases for parse login and remove validity in ParserTest.java

manuelmanuntag96 committed:
Added more test cases for LogInCommand in LogInCommandTest.java

manuelmanuntag96 committed:
Edited Remove User Command to new format: remove /username /password

manuelmanuntag96 committed:
Edited to comply with gradlew checkstyle

manuelmanuntag96 committed:
removed printl for debugging

yellow-fellow committed:
Updated Compare Command overall UX

tammykoh committed:
Removed case-sensitivity for day

yellow-fellow committed:
Updated error handling exception

tammykoh committed:
Clarified error message for wrong day

yeapcl committed:
Merge pull request #196 from yellow-fellow/master

Updated Compare Command overall UX

yeapcl committed:
Merge pull request #195 from manuelmanuntag96/Manuel-RemoveWithPassword

Manuel remove with password

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Updated error handling exception
  Updated Compare Command overall UX
  removed printl for debugging
  Edited to comply with gradlew checkstyle
  Edited Remove User Command to new format: remove /username /password
  Added more test cases for LogInCommand in LogInCommandTest.java
  Added even more test cases for parse login and remove validity in ParserTest.java
  Added more test cases for parse login and remove validity in ParserTest.java
  Updates to User Guide and added more Junit test
  Updated Userguide and add more Junit testing
  Added more test cases for UserListTest.java
  Updates the About Us and the table of content for User Guide
  Update UserGuide.md
  Updates to Developer Guide
  Update to UserGuide command summary
  Updates to UserGuide
  Implemented the Remove Feature
  Updates to Userguid expected outputs and command formats corrections
  Updated Expected outputs for Edit,list,login,find,delete,clear,exit Commands in the Userguide
  improved EditCommand where if new timings and original timings are identical, program would prompt user.

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Updated error handling exception
  Updated Compare Command overall UX
  removed printl for debugging
  Edited to comply with gradlew checkstyle
  Edited Remove User Command to new format: remove /username /password
  Added more test cases for LogInCommand in LogInCommandTest.java
  Added even more test cases for parse login and remove validity in ParserTest.java
  Added more test cases for parse login and remove validity in ParserTest.java
  Updates to User Guide and added more Junit test
  Updated Userguide and add more Junit testing
  Added more test cases for UserListTest.java
  Updates the About Us and the table of content for User Guide
  Update UserGuide.md
  Updates to Developer Guide
  Update to UserGuide command summary
  Updates to UserGuide
  Implemented the Remove Feature
  Updates to Userguid expected outputs and command formats corrections
  Updated Expected outputs for Edit,list,login,find,delete,clear,exit Commands in the Userguide

yeapcl committed:
Improved CompareCommand such the current user's name would not appear in the userlist, while preserving its true arraylist index.

yeapcl committed:
Further improved user-friendliness of UI with more concise instructions

yeapcl committed:
fixed checkstyle and updated expected.txt for new UI messages

weisiong24 committed:
Merge pull request #198 from yeapcl/branch-yeapcl-enhanceProgramUsability

Improved CompareCommand and UI

tammykoh committed:
Merge remote-tracking branch 'upstream/master' into tammy-fixing-bugs

tammykoh committed:
Merge branch 'tammy-fixing-bugs'

tammykoh committed:
Fixed checkstyle error

yeapcl committed:
Merge pull request #199 from tammykoh/master

Fixed bugs from PE

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Editted the basics of Manuel's PPP

manuelmanuntag96 committed:
Edited Manuel's PPP

manuelmanuntag96 committed:
Merge pull request #200 from manuelmanuntag96/Manuel-PPP

Manuel ppp

yeapcl committed:
Revamped EditCommand UI to improve user-friendliness
Implemented on-screen prompt for user so that it's more intuitive for user to edit a lesson
Further enhanced time validity check.
Edited timings must now be in 30-minute block

weisiong24 committed:
Merge pull request #202 from yeapcl/branch-yeapcl-enhanceProgramUsability

Revamped EditCommand UI to improve user-friendliness

Commit:
Updated Add Command checks

Commit:
Updates to add command checks

yeapcl committed:
Merge pull request #203 from weisiong24/BugFixes

Updated the check for 1hour block in Add Command

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Updates to add command checks
  Updated Add Command checks
  Revamped EditCommand UI to improve user-friendliness Implemented on-screen prompt for user so that it's more intuitive for user to edit a lesson Further enhanced time validity check. Edited timings must now be in 30-minute block
  Edited Manuel's PPP
  Editted the basics of Manuel's PPP
  Fixed checkstyle error
  fixed checkstyle and updated expected.txt for new UI messages
  Further improved user-friendliness of UI with more concise instructions
  Improved CompareCommand such the current user's name would not appear in the userlist, while preserving its true arraylist index.
  Clarified error message for wrong day
  Removed case-sensitivity for day

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Updates to add command checks
  Updated Add Command checks
  Edited Manuel's PPP
  Editted the basics of Manuel's PPP
  Fixed checkstyle error
  Clarified error message for wrong day
  Removed case-sensitivity for day

yeapcl committed:
edited timings are now restricted to 1-hour block

Commit:
Updated the error messages for the add commands

weisiong24 committed:
Merge pull request #204 from yeapcl/branch-yeapcl-enhanceProgramUsability

Limit timings to 1-hour block

yeapcl committed:
Merge pull request #205 from weisiong24/BugFixes

Updated the error messages for the add commands

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Updated the error messages for the add commands
  edited timings are now restricted to 1-hour block

yeapcl committed:
Merge branch 'master' into branch-yeapcl-enhanceProgramUsability

* master:
  Updated the error messages for the add commands

Commit:
Updated UserGuide,Developer Guide for updated edit command and Added my PPP

weisiong24 committed:
Merge pull request #206 from weisiong24/BugFixes

Updated UserGuide,Developer Guide for updated edit command and Added …

Commit:
Updated ppp

weisiong24 committed:
Merge pull request #207 from weisiong24/BugFixes

Updated ppp

Commit:
Update ppp

weisiong24 committed:
Merge pull request #208 from weisiong24/BugFixes

Update ppp

Commit:
Updates ppp

weisiong24 committed:
Merge pull request #209 from weisiong24/BugFixes

Updates ppp

yeapcl committed:
Added more exception types
Added more JUnit test for 100% test coverage for EditCommand

yeapcl committed:
edited outputstream for windows CI test

yeapcl committed:
minor fix for checkstyle

yeapcl committed:
modified text formatting for windows gradle junit test

yeapcl committed:
edited expected runtest results

yeapcl committed:
fix outputstream for windows

Commit:
Updates profie pic

weisiong24 committed:
Merge pull request #211 from weisiong24/BugFixes

Updates profie pic

Commit:
Updates ppp

weisiong24 committed:
Merge pull request #212 from weisiong24/BugFixes

Updates ppp

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

tammykoh committed:
Merge pull request #210 from yeapcl/branch-yeapcl-JUnitTest

JUnit tests for EditCommand

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Added more to Manuel's PPP

manuelmanuntag96 committed:
Merge pull request #213 from manuelmanuntag96/Manuel-PPP2

Manuel ppp2

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Update manuelmanuntag96.md

manuelmanuntag96 committed:
Update manuelmanuntag96.md

manuelmanuntag96 committed:
Update manuelmanuntag96.md

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Update manuelmanuntag96.md

yellow-fellow committed:
Updated PPP

yellow-fellow committed:
Updated PPP

tammykoh committed:
Merge branch 'master' into tammy-fixing-bugs

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into branch-yeapcl-JUnitTest

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Updated PPP
  Updated PPP
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Added more to Manuel's PPP
  Updates ppp
  Updates profie pic
  Updates ppp
  Update ppp
  Updated ppp
  Updated UserGuide,Developer Guide for updated edit command and Added my PPP

tammykoh committed:
Created 'clear /all' function

tammykoh committed:
Editted comments for clear command

yeapcl committed:
bug fix

tammykoh committed:
Added 'clear /all' into UG
Fixed spelling error

tammykoh committed:
Merge branch 'master' into tammy-developer-guide

yeapcl committed:
bug fix

tammykoh committed:
Added activation bar for UI in find, list, delete, clear sequence diagrams

tammykoh committed:
Merge branch 'tammy-fixing-bugs'

tammykoh committed:
Merge branch 'tammy-developer-guide'

yeapcl committed:
bug fix

yeapcl committed:
bug fix

yeapcl committed:
bug fix

yeapcl committed:
format fix attempt

yeapcl committed:
format fix attempt

yeapcl committed:
finally found the elusive bug

tammykoh committed:
Added PPP

tammykoh committed:
Merge branch 'tammy-ppp'

tammykoh committed:
Merge branch 'master' into tammy-testing

yeapcl committed:
junit test for EditCommand 100% line coverage and successful on gradlew windows test

yeapcl committed:
JUnit for EditCommand

100% line coverage and fixed windows gradlew check

tammykoh committed:
Added more test for ClearCommand

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into branch-yeapcl-enhanceProgramUsability

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp: (28 commits)
  junit test for EditCommand 100% line coverage and successful on gradlew windows test
  finally found the elusive bug
  format fix attempt
  format fix attempt
  bug fix
  bug fix
  bug fix
  bug fix
  bug fix
  Updated PPP
  Updated PPP
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Added more to Manuel's PPP
  Updates ppp
  Updates profie pic
  fix outputstream for windows
  edited expected runtest results
  ...

yeapcl committed:
code clean up

yeapcl committed:
format fix

yeapcl committed:
Merge pull request #215 from yeapcl/branch-yeapcl-enhanceProgramUsability

Code cleanup and added Storage initialisation messages

yeapcl committed:
further code clean up.
removed TestCommand class meant for internal developer test.

yeapcl committed:
slightly improved storage initialisation messages

yeapcl committed:
Merge pull request #216 from yeapcl/branch-yeapcl-enhanceProgramUsability

Further code clean up

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into branch-yeapcl-Documentation

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp: (80 commits)
  slightly improved storage initialisation messages
  further code clean up. removed TestCommand class meant for internal developer test.
  format fix
  code clean up
  junit test for EditCommand 100% line coverage and successful on gradlew windows test
  finally found the elusive bug
  format fix attempt
  format fix attempt
  bug fix
  bug fix
  bug fix
  bug fix
  bug fix
  Updated PPP
  Updated PPP
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Update manuelmanuntag96.md
  Added more to Manuel's PPP
  ...

tammykoh committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

tammykoh committed:
Merge branch 'master' into tammy-testing

tammykoh committed:
Included more test for ClearCommand to increase test coverage

tammykoh committed:
Added tests for DeleteCommand

tammykoh committed:
Added test for FindCommand

tammykoh committed:
Added test for ListCommand

tammykoh committed:
Added test to Parser for list, find, clear and delete commands

tammykoh committed:
Merge branch 'tammy-testing'

tammykoh committed:
Fixed checkstyle error

tammykoh committed:
Updated expected output

yeapcl committed:
Merge pull request #217 from tammykoh/master

Fixed Issues and Added Test

yeapcl committed:
updated PPP and UG

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into branch-yeapcl-Documentation

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Updated expected output
  Fixed checkstyle error
  Added test to Parser for list, find, clear and delete commands
  Added test for ListCommand
  Added test for FindCommand
  Added tests for DeleteCommand
  Included more test for ClearCommand to increase test coverage
  Added more test for ClearCommand
  Added PPP
  Added activation bar for UI in find, list, delete, clear sequence diagrams
  Added 'clear /all' into UG Fixed spelling error
  Editted comments for clear command
  Created 'clear /all' function

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Added Remove Feature Details in UG

manuelmanuntag96 committed:
Merge pull request #218 from manuelmanuntag96/Manuel-RemoveUserUG

Manuel remove user ug

manuelmanuntag96 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

manuelmanuntag96 committed:
Edited Manuel's PPP part 3

manuelmanuntag96 committed:
Merge pull request #219 from manuelmanuntag96/Manuel-PPP3

Manuel ppp3

yeapcl committed:
further update to PPP

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into branch-yeapcl-Documentation

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Edited Manuel's PPP part 3
  Added Remove Feature Details in UG

yeapcl committed:
Merge pull request #220 from yeapcl/branch-yeapcl-Documentation

PPP and UG

manuelmanuntag96 committed:
Update manuelmanuntag96.md

yeapcl committed:
further update to UG and PPP

yeapcl committed:
Merge pull request #221 from yeapcl/branch-yeapcl-Documentation

further update to UG and PPP

manuelmanuntag96 committed:
Update UserGuide.md

yeapcl committed:
updated test cases and UG

tammykoh committed:
Added logging for list, clear, find and delete functions

tammykoh committed:
Merge branch 'master' into tammy-logging

tammykoh committed:
Merge branch 'tammy-logging'

tammykoh committed:
Merge branch 'master' into tammy-testing

yeapcl committed:
added more javadoc and improved UG

yeapcl committed:
minor formatting fix

yeapcl committed:
formatting fix

yeapcl committed:
Merge pull request #222 from yeapcl/branch-yeapcl-Documentation

Updated UG, javadoc and logging

tammykoh committed:
Added test for compare command

tammykoh committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into tammy-testing

tammykoh committed:
Merge branch 'master' into tammy-testing

tammykoh committed:
Merge branch 'tammy-testing'

tammykoh committed:
Fixed checkstyle
Organised files

tammykoh committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp

tammykoh committed:
Merge pull request #223 from tammykoh/master

Added Logging

tammykoh committed:
Deleted temporary storage folder
Updated About Us

yeapcl committed:
further documentation update

yeapcl committed:
update DG

yeapcl committed:
further update DG

yeapcl committed:
Merge pull request #225 from yeapcl/branch-yeapcl-Documentation

further documentation update

yeapcl committed:
Merge pull request #224 from tammykoh/master

Minor updates

yeapcl committed:
updated JAR link to 2.1

yeapcl committed:
Merge pull request #226 from yeapcl/branch-yeapcl-Documentation

updated JAR link to 2.1

tammykoh committed:
Updated JAR link

yeapcl committed:
Merge pull request #227 from tammykoh/master

Updated JAR link

yeapcl committed:
format fix

yeapcl committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into branch-yeapcl-Documentation

* 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp:
  Updated JAR link
  Deleted temporary storage folder Updated About Us
  Fixed checkstyle Organised files
  Merge branch 'master' of https://github.com/AY2021S1-CS2113-T13-3/tp into tammy-testing
  Added test for compare command
  Added logging for list, clear, find and delete functions
  Update UserGuide.md
  Update manuelmanuntag96.md

yeapcl committed:
Merge pull request #228 from yeapcl/branch-yeapcl-Documentation

DG formatting fix
